process/block/interceptedBlocks/interceptedEquivalentProof.go

```diff
@@ -31,6 +31,7 @@ type ArgInterceptedEquivalentProof struct {
 	HeaderSigVerifier consensus.HeaderSigVerifier
 	Proofs            dataRetriever.ProofsPool
 	Headers           dataRetriever.HeadersPool
+	KeyRWMutexHandler sync.KeyRWMutexHandler
```

Comment:
Do we have to receive it as a parameter, is it a shared resource? If it's only an internal component that helps us serialize some proof checks, we can simply instantiate it here, in the constructor.